apiserver/facades/client/client/status.go

```diff
@@ -1535,7 +1537,7 @@ func (context *statusContext) unitPublicAddress(unit *state.Unit) string {
 
 func (context *statusContext) processUnit(ctx context.Context, unit *state.Unit, applicationCharm string) params.UnitStatus {
 	var result params.UnitStatus
-	if prs, ok := context.allOpenPortRanges[unit.Tag().Id()]; ok {
+	if prs, ok := context.allOpenPortRanges[coreunit.Name(unit.Name())]; ok {
```

Comment:
**todo:** No validation with coreunint.NewName() ? If it is normal, we should at least add a comment which explain why.

Reply:
By logic was, since `unit.Name()` queries a unit name in Mongo, we can trust it. But there's no harm in double checking

Reply:
This will naturally resolve itself when we've removed Mongo from this endpoint. Since domain methods will naturally return `unit.Names`, meaning the apiserver won't need to worry about casting data types

Reply:
Ok, thanks, I can live with that explanation !